test/e2e/index.ts

```diff
@@ -422,6 +422,28 @@ test(
   })
 );
 
+test(
+  "invalid app authentication token fails",
+  withConductor(ADMIN_PORT, async (t) => {
+    const { admin } = await installAppAndDna(ADMIN_PORT);
+    const { port } = await admin.attachAppInterface({
+      allowed_origins: "client-test-app",
+    });
+    try {
+      await AppWebsocket.connect({
+        url: new URL(`ws://localhost:${port}`),
+        wsClientOptions: { origin: "client-test-app" },
+        token: [0],
+      });
+      t.fail("could connect with invalid authentication token");
+    } catch (error) {
+      t.assert(error instanceof HolochainError);
+      assert(error instanceof HolochainError);
+      t.equal(error.name, "InvalidTokenError", "expected InvalidTokenError");
+    }
```

Comment:
If at this point you call it again, I'd expect to get a "token is required", rather than "invalid token error"?

Reply:
This only calls once. What are you referring to by "you call it again"?

Reply:
I see now this is the wrong test to ask this question. I'm looking for a test where we
- connect with a valid token that is single use,
- lose the connection then try to make a zome call, but now have an invalid token, 
- so we get a invalid token error
- then try to make another zome call and we should get a "no token" error or similar because the reconnect can't be attempted

Reply:
Is that what the test below with the comment "Websocket reconnection has failed and subsequent calls should just return" is doing?

Reply:
Sweet, okay. I'm caught up now and my original concern with this change has been addressed. Thank you :)